adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,545 @@
+ * corresponding to the arrangement of the two panes of the layout.
+ *
+ * This strategy can be conditional: If `null` is returned from [calculateSplitResult], then this
+ * strategy did not produce a split result to use, and a different strategy should be used.
+ */
+private fun interface ConditionalTwoPaneStrategy {
+    /**
+     * Calculates the split result in local bounds of the [TwoPane], or `null` if this strategy
+     * does not apply.
+     *
+     * @param density the [Density] for measuring and laying out
+     * @param layoutDirection the [LayoutDirection] for measuring and laying out
+     * @param layoutCoordinates the [LayoutCoordinates] of the [TwoPane]
+     */
+    public fun calculateSplitResult(
+        density: Density,
+        layoutDirection: LayoutDirection,
+        layoutCoordinates: LayoutCoordinates
+    ): SplitResult?
+}
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots vertically or horizontally if there is a
+ * horizontal or vertical fold respectively.
+ *
+ * If there is no fold, then the [defaultStrategy] will be used instead.
+ */
+public fun TwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    defaultStrategy: TwoPaneStrategy,
+): TwoPaneStrategy = TwoPaneStrategy(
+    FoldAwareHorizontalTwoPaneStrategy(windowGeometry),
+    FoldAwareVerticalTwoPaneStrategy(windowGeometry),
+    defaultStrategy = defaultStrategy
+)
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally if there is a vertical fold.
+ *
+ * If there is no fold, then the [defaultStrategy] will be used instead.
+ */
+public fun HorizontalTwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    defaultStrategy: TwoPaneStrategy,
+): TwoPaneStrategy = TwoPaneStrategy(
+    FoldAwareHorizontalTwoPaneStrategy(windowGeometry),
+    defaultStrategy = defaultStrategy
+)
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally.
+ *
+ * If there is a vertical fold, then the gap will be placed along the fold.
+ *
+ * Otherwise, the gap will be placed at the given [splitFraction] from start, with the given
+ * [gapWidth].
+ */
+public fun HorizontalTwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    splitFraction: Float,
+    gapWidth: Dp = 0.dp,
+): TwoPaneStrategy = TwoPaneStrategy(
+    FoldAwareHorizontalTwoPaneStrategy(windowGeometry),
+    defaultStrategy = FractionHorizontalTwoPaneStrategy(
+        splitFraction = splitFraction,
+        gapWidth = gapWidth
+    )
+)
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally.
+ *
+ * If there is a vertical fold, then the gap will be placed along the fold.
+ *
+ * Otherwise, the gap will be placed at [splitOffset] either from the start or end based on
+ * [offsetFromStart], with the given [gapWidth].
+ */
+public fun HorizontalTwoPaneStrategy(
+    windowGeometry: WindowGeometry,
+    splitOffset: Dp,
+    offsetFromStart: Boolean,
```

Comment:
Seems like this might have a reasonable default, perhaps, `true`? 

Also, I cannot understand why, but having this boolean makes me want to split the functions into `FixedStartTwoPaneStrategy(orientation:Orientation, ...)` and `FixedEndTwoPaneStrategy`.

Reply:
Updated to have default of `true`!